main.tf

```diff
@@ -147,6 +161,20 @@ resource "aws_cloudfront_distribution" "cloudfront_distribution" {
         }
         headers = ordered_cache_behavior.value.headers
```

Comment:
headers = "${compact(ordered_cache_behavior.value.headers != "" ? ordered_cache_behavior.value.headers : list(""))}"

in case we are don't need to have any headers, when we set the headers = [""] at the variable map, the assignment fails with error saying headers cant be empty as there is no value to the list.
Fixed the above with this interpolation.